libs/designsystem/shared/floating/src/floating.directive.spec.ts

```diff
@@ -0,0 +1,463 @@
+import { Component, ElementRef, Input, ViewChild } from '@angular/core';
+import { createComponentFactory, mockProvider, Spectator } from '@ngneat/spectator';
+import { PortalDirective } from '@kirbydesign/designsystem/shared/portal';
+import { TriggerEvent } from '@kirbydesign/designsystem/shared/floating';
+import { FloatingDirective } from './floating.directive';
+
+import * as floatingUi from '@floating-ui/dom';
+import { Placement } from '@floating-ui/core/src/types';
+import { Strategy } from '@floating-ui/dom';
+import any = jasmine.any;
+import { DesignTokenHelper } from '@kirbydesign/core';
+
+@Component({
+  template: `
+    <div #floatingElement></div>
+    <div
+      #hostElement
+      kirbyFloating
+      [reference]="floatingElementRef"
+      [isDisabled]="isDisabled"
+      [closeOnEscapeKey]="closeOnEscapeKey"
+      [closeOnSelect]="closeOnSelect"
+      [closeOnBackdrop]="closeOnBackdrop"
+      [strategy]="strategy"
+      [placement]="placement"
+    ></div>
+    <div #clickTarget></div>
+  `,
+  imports: [FloatingDirective],
+  standalone: true,
+})
+class FloatingTestComponent {
+  @ViewChild('floatingElement', { static: true }) public floatingElementRef: ElementRef;
+  @ViewChild('hostElement', { static: true }) public hostElementRef: ElementRef;
+  @ViewChild('clickTarget', { static: true }) public clickTargetRef: ElementRef;
+  @ViewChild(FloatingDirective, { static: true }) public floatingDirective: FloatingDirective;
+
+  /** Start values set to match directive */
+  @Input() public isDisabled: boolean = false;
+  @Input() public closeOnEscapeKey: boolean = true;
+  @Input() public closeOnSelect: boolean = true;
+  @Input() public closeOnBackdrop: boolean = true;
+  @Input() public strategy: Strategy = 'absolute';
+  @Input() public placement: Placement = 'bottom-start';
+}
```

Comment:
Cant we just not initialise them, so is uses the default values from the directive, and then you set the input when needed?

Reply:
Since I bind them to the directive, setting them to undefined would also give that value to the directive, thus not being able to use the default values. 

I need these inputs to be able to simulate actually changing the values from the component using the directive :)

Reply:
I see 👍